File: recall/positive/python/log_injection.py
"""Log injection TP fixtures: AST-LOG-001 must fire."""
import logging
from flask import request

logger = logging.getLogger(__name__)


def handler_stdlib():
    user = request.args.get('user')
    logging.info("Login: " + user)      # AST-LOG-001 MEDIUM


def handler_logger_instance():
    action = request.form.get('action')
    logger.warning(action)              # AST-LOG-001 MEDIUM

File: recall/positive/java/log_injection.java
import javax.servlet.http.HttpServletRequest;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Java log injection TP: JAST-LOG-001 must fire.
 */
public class LogInjectionTest {
    private static final Logger logger = LoggerFactory.getLogger(LogInjectionTest.class);

    public void handle(HttpServletRequest request) {
        String user = request.getParameter("user");
        logger.info("Login attempt: " + user);   // JAST-LOG-001 MEDIUM
    }
}
